src/vg-tooltip.js

```diff
@@ -60,34 +79,55 @@
     // initialize tooltip with item data and options on mouse over
     vgView.on("mouseover.tooltipInit", function(event, item) {
       if (shouldShowTooltip(item)) {
-        init(event, item, options);
+        // clear existing promise because mouse can only point at one thing at a time
+        cancelPromise();
+
+        // make a new promise with time delay for tooltip
+        tooltipPromise = window.setTimeout(function() {
+          init(event, item, options);
+          tooltipActive = true;
+        }, options.delay || DELAY);
       }
     });
 
     // update tooltip position on mouse move
     // (important for large marks e.g. bars)
     vgView.on("mousemove.tooltipUpdate", function(event, item) {
-      if (shouldShowTooltip(item)) {
+      if (shouldShowTooltip(item) && tooltipActive) {
         update(event, item, options);
       }
     });
 
     // clear tooltip on mouse out
     vgView.on("mouseout.tooltipClear", function(event, item) {
       if (shouldShowTooltip(item)) {
-        clear(event, item, options);
+        cancelPromise();
+
+        if (tooltipActive) {
+          clear(event, item, options);
+          tooltipActive = false;
+        }
       }
     });
 
     return {
       destroy: function() {
+        // remove event listeners
         vgView.off("mouseover.tooltipInit");
         vgView.off("mousemove.tooltipUpdate");
         vgView.off("mouseout.tooltipClear");
+
+        cancelPromise(); // clear tooltip promise
       }
     }
   };
 
+  // cancel tooltip promise
+  function cancelPromise() {
+    window.clearTimeout(tooltipPromise);
```

Comment:
// Passing an invalid ID to clearTimeout does not have any effect (and doesn't throw an exception).